js/components/common/react_modal.js.jsx

```diff
@@ -34,19 +34,86 @@ class CourseModal extends React.Component {
       course: [],
       sessions: [],
       courseTitle: "",
+      clicked: 0,
     }
   }
+  /**
+   * Change the clicked and courseId state, whenever a course link is clicked.
+   */
+  clickRelated = s => {
+    let copyState = this.state
+    copyState.courseId = s
+    copyState.clicked = 1
+    this.setState(copyState)
+  }
+
+  /**
+   * Convert the course names to <a> tags and return a list of strings and <a> tags.
+   */
+  getRelatedCourses = content => {
+    const result = []
+    if (content !== null) {
+      content.split(" ").forEach((word, i) => {
+        if (word.match(/[A-Z]{3,4}[0-9]{3}[H|Y]1/gi)) {
+          let symbol = ""
+          // check if the last character is not 1 (is a symbol). If so, remove it from word and
```

Comment:
Don't split up the string. As I've said before, your code should work correctly passing in the complete course code, including the H1/Y1.

Reply:
@david-yz-liu The splitting by space is done for the entire string, not the course code. So the course code would be for example 'CSC207H1' . Should I still keep the split?

Reply:
Here I meant the splitting of the "H1"/"Y1" is unnecessary (but splitting on spaces is good!).